scripts/run

```diff
@@ -0,0 +1,127 @@
+                  either `local` or `docker`
+- local           run as a local terminal process
+- docker          run in a docker container
+- check           run linter and checks
+EOF
+}
+
+# script variables and defaults
+source "$(dirname "$0")"/init.sh
+
+AZURE_ENV=dev
+
+# process command-line options (if any)
+die() { log_error "$*" >&2; echo ""; usage; exit 2; }  # complain to STDERR and exit with error
+needs_arg() { if [ -z "$OPTARG" ]; then die "No arg for --$OPT option"; fi; }
+
+while getopts h-: OPT; do
+  # support long options: https://stackoverflow.com/a/28466267/519360
+  if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
+    OPT="${OPTARG%%=*}"       # extract long option name
+    OPTARG="${OPTARG#"$OPT"}"   # extract long option argument (may be empty)
+    OPTARG="${OPTARG#=}"      # if long option argument, remove assigning `=`
+  fi
+  case "$OPT" in
+    h | help )     usage; exit 0 ;;
+    \? )           usage; exit 2 ;;  # bad short option (error reported via getopts)
+    * )            die "Illegal option --$OPT" ;;            # bad long option
+  esac
+done
+shift $((OPTIND-1)) # remove parsed options and args from $@ list
+
+
+# process positional arguments (if any)
+RUNTYPE=${1}
+
+run_tests() {
+  eval "$("${SCRIPTS_DIR}"/render-configs.sh -a "$AZURE_ENV" -e)"
+  cd "${ROOT_DIR}"
+  # shellcheck disable=SC2086 # The array syntax and quoting seems to break gradle
+  ./gradlew ${GRADLE_OPTS} testUnit jacocoTestReport
+
+#./gradlew --build-cache srcclr
+}
+
+run_integration_tests() {
+  # NOTE: interesting - no integration-tests.yml have been triggered or run
+  # this is run via both:
+  # - ./github/workflows/build-and-test.yml
+  # - ./github/workflows/integration-tests.yml
+  #
+  # please see integration-tests.yml for more information re: inputs.environment and TEST_ENV
+  TEST_ENV=${TEST_ENV:-local}
+
+  cd "${ROOT_DIR}"
+  export IT_JADE_API_URL=http://localhost:8080
+  AZURE_ENV=integration
+  run_docker
+  # shellcheck disable=SC2086 # The array syntax and quoting seems to break gradle
+  ./gradlew ${GRADLE_OPTS} testIntegration
+
+  # for developer runs, build an integration report locally
+  if [ -z "$CI" ]; then
+    log_debug "TODO: how do we build a report?"
+  fi
+}
+
+run_connected_tests() {
+  # NOTE: interesting - no integration-tests.yml have been triggered or run
+  # this is run via both:
+  # - ./github/workflows/build-and-test.yml
+  # - ./github/workflows/integration-tests.yml
+  #
+  # please see integration-tests.yml for more information re: inputs.environment and TEST_ENV
+  TEST_ENV=${TEST_ENV:-local}
+
+  cd "${ROOT_DIR}"
+  export IT_JADE_API_URL=http://localhost:8080
+  run_docker
+  # shellcheck disable=SC2086 # The array syntax and quoting seems to break gradle
+  ./gradlew ${GRADLE_OPTS} testConnected
+
+  # for developer runs, build an integration report locally
+  if [ -z "$CI" ]; then
+    log_debug "TODO: how do we build a report?"
+  fi
+}
+
+run_local() {
```

Comment:
This script uses `run` to mean two different things. In this case, `run_local` starts a running copy of the service. Whereas `run_test` runs the unit tests. Maybe this should be `start_local` and `start_docker`?